### PR TITLE
`azurerm_mysql_*` - deprecate Mysql Single Server and its sub resources

### PR DESCRIPTION
**Pull request by neil-yechenwei**
Service team is reaching us to add this warning for deprecating Mysql Single Server and its sub resources.

### CODE REVIEW

File: internal/services/mysql/mysql_aad_administrator_resource.go [hunk 1]
```diff
@@ -25,6 +25,9 @@ func resourceMySQLAdministrator() *pluginsdk.Resource {
 		Read:   resourceMySQLAdministratorRead,
 		Update: resourceMySQLAdministratorCreateUpdate,
 		Delete: resourceMySQLAdministratorDelete,
+
+		DeprecationMessage: "The `azurerm_mysql_active_directory_administrator` resource will be deprecated since Azure Database for MySQL - Single Server and its sub resources are scheduled for retirement (https://go.microsoft.com/fwlink/?linkid=2216041) by September 16, 2024 and they migrate (https://go.microsoft.com/fwlink/?linkid=2202255) to Azure Database for MySQL - Flexible Server and its sub resources now. Please use the `azurerm_mysql_flexible_server_active_directory_administrator` resource instead.",
```
Comment (stephybun) [inline internal/services/mysql/mysql_aad_administrator_resource.go:29]:
```suggestion
		DeprecationMessage: "Azure Database for MySQL Single Server and its sub resources are scheduled for retirement by 2024-09-16 and will migrate to using Azure Database for MySQL Flexible Server: https://go.microsoft.com/fwlink/?linkid=2216041. The `azurerm_mysql_active_directory_administrator` resource is deprecated and will be removed in v4.0 of the AzureRM Provider. Please use the `azurerm_mysql_flexible_server_active_directory_administrator` resource instead.",
```

File: internal/services/mysql/mysql_configuration_resource.go [hunk 2]
```diff
@@ -24,6 +24,8 @@ func resourceMySQLConfiguration() *pluginsdk.Resource {
 		Read:   resourceMySQLConfigurationRead,
 		Delete: resourceMySQLConfigurationDelete,
 
+		DeprecationMessage: "The `azurerm_mysql_configuration` resource will be deprecated since Azure Database for MySQL - Single Server and its sub resources are scheduled for retirement (https://go.microsoft.com/fwlink/?linkid=2216041) by September 16, 2024 and they migrate (https://go.microsoft.com/fwlink/?linkid=2202255) to Azure Database for MySQL - Flexible Server and its sub resources now. Please use the `azurerm_mysql_flexible_server_configuration` resource instead.",
```
Comment (stephybun) [inline internal/services/mysql/mysql_configuration_resource.go:27]: Can you please apply the suggestion made above to all of the deprecation messages in this PR

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @neil-yechenwei. I left a suggestion on the phrasing of the deprecation message, could you please apply those changes to the rest of the deprecation messages?

File: internal/services/mysql/mysql_aad_administrator_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/mysql/mysql_aad_administrator_resource.go]: Updated

File: internal/services/mysql/mysql_configuration_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/mysql/mysql_configuration_resource.go]: Updated to all.

**Comment by neil-yechenwei** [comment]
@stephybun , thanks for the comments. I updated PR. Please take another look. Thanks.

**Review by stephybun — APPROVED** [review]
Thanks @neil-yechenwei LGTM 👍

**Pull request merged**